src/Symfony/Component/Messenger/Transport/KernelTerminate/KernelTerminateTransport.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\KernelTerminate;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+class KernelTerminateTransport implements TransportInterface
+{
+    private $handlers = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function receive(callable $handler): void
+    {
+        $this->handlers[] = $handler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function stop(): void
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function send(Envelope $envelope)
+    {
+    }
+
+    public function __invoke()
```

Comment:
I guess it's the kernel.terminate listener ? :) 
I won't use an invokable for this. Perhaps `flush` ?